From 9f02996b12fb90b3eac44c0f3c3b964fe15a3728 Mon Sep 17 00:00:00 2001
From: Ian Dannapel <ian.dannapel@iris-sensing.com>
Date: Tue, 3 May 2022 16:26:02 +0200
Subject: [PATCH] tc358746.c: Reset when changing format

To actually change the format, a software reset is needed.
Since the capture driver does not do that, we force it.

Signed-off-by: Ian Dannapel <ian.dannapel@iris-sensing.com>
---
 drivers/media/i2c/tc358746.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/media/i2c/tc358746.c b/drivers/media/i2c/tc358746.c
index 94bf616d03f0..ba5bacde966d 100644
--- a/drivers/media/i2c/tc358746.c
+++ b/drivers/media/i2c/tc358746.c
@@ -1299,7 +1299,8 @@ static int tc358746_set_fmt(struct v4l2_subdev *sd,
 			v4l2_ctrl_s_ctrl(state->link_freq, new_freq);
 	}
 
-	tc358746_set_buffers(sd);
+	tc358746_s_power(sd, 0);
+	tc358746_s_power(sd, 1);
 	return 0;
 }
 
-- 
2.39.1

